primitives/runtime/src/generic/unchecked_extrinsic.rs

```diff
@@ -48,6 +49,25 @@ where
 	pub function: Call,
 }
 
+/// Manual [`TypeInfo`] implementation because of custom encoding. The data is a valid encoded
+/// `Vec<u8>`, but requires some logic to extract the signature and payload.
+///
+/// See [`UncheckedExtrinsic::encode`] and [`UncheckedExtrinsic::decode`].
+impl<Address, Call, Signature, Extra> TypeInfo
+	for UncheckedExtrinsic<Address, Call, Signature, Extra>
+where
+	Extra: SignedExtension + TypeInfo,
+{
+	type Identity = UncheckedExtrinsic<(), (), (), Extra>;
```

Comment:
Is it correct to write it like this ? because the types `UncheckedExtrinsic<(), (), (), Extra>;` and `UncheckedExtrinsic<Address, Call, Signature, Extra>` are different.
Could that lead to invalid type info ? if not I would prefer having an inner comment telling why it is correct.

Reply:
I had done that intentionally because the fields with those types are opaque, since because of the custom encoding it can only be described as a raw `Vec<u8>`. And so they would all effectively have the same type.

However, on reflaction I think it can be improved by adding those type parameters anyway, which will aid the downstream consumers in the manual decoding of the actual fields from the raw bytes. Let me know what you think.

Reply:
sounds good to me now